internal/js_scanner/js_scanner.go

```diff
@@ -95,22 +96,31 @@ outer:
 				i += len(nextValue)
 				flags[string(nextValue)] = true
 
+				if next == js.ErrorToken && l.Err() == io.EOF {
+					isEOF = true
+				}
+
 				if js.IsIdentifier(next) {
-					if isKeyword(nextValue) && next != js.FromToken {
+					if string(nextValue) == "type" {
+						flags["type"] = true
 						continue
 					}
-					if string(nextValue) == "type" {
+					if isKeyword(nextValue) && next != js.FromToken {
 						continue
 					}
 					if !foundIdent {
 						foundIdent = true
 					}
-				} else if next == js.LineTerminatorToken || next == js.SemicolonToken || (next == js.ErrorToken && l.Err() == io.EOF) {
+				} else if next == js.LineTerminatorToken || next == js.SemicolonToken || isEOF {
 					if (flags["function"] || flags["=>"] || flags["interface"]) && !flags["{"] {
-						continue
+						if !isEOF {
```

Comment:
can these if blocks be simplified somehow or does this make the most sense?

Reply:
Oh yeah, not sure what I was thinking. Cleaned it up significantly!

Reply:
nice job!